src/io/ipc/read/read_basic.rs

```diff
@@ -105,9 +107,8 @@ fn read_compressed_buffer<T: NativeType, R: Read + Seek>(
     let mut buffer = vec![T::default(); length];
 
     // decompress first
-    // todo: move this allocation to an external buffer for re-use
-    let mut slice = vec![0u8; buffer_length];
-    reader.read_exact(&mut slice)?;
+    scratch.set_len(buffer_length);
```

Comment:
Had to add scratch to most functions to reach this point. Here we can reuse the scratch for every decompression.